### PR TITLE
Support typevar type check

### PR DESCRIPTION
**Pull request by uriva**

### CODE REVIEW

File: gamla/functional.py [hunk 1]
```diff
@@ -169,9 +169,9 @@ def compute_stable_json_hash(item) -> Text:
 def assert_that_with_message(input_to_message: Callable, f: Callable):
     """Assert a function `f` on the input, printing the output of `input_to_message(input)` if assertion is False.
 
-    >>> assert_that_with_message(gamla.just("Input is not 2!"), operator.equals(2))(2)
+    >>> assert_that_with_message(gamlaconstruct.just("Input is not 2!"), operator.equals(2))(2)
```
Comment (itayzit) [inline gamla/functional.py:172]: construct.just

File: gamla/functional.py [hunk 2]
```diff
@@ -169,9 +169,9 @@ def compute_stable_json_hash(item) -> Text:
...
     2
-    >>> assert_that_with_message(gamla.just("Input is not 2!"), operator.equals(2))(3)
+    >>> assert_that_with_message(gamlaconstruct.just("Input is not 2!"), operator.equals(2))(3)
```
Comment (itayzit) [inline gamla/functional.py:174]: construct.just

File: gamla/functional.py (hunk 1 above)
Comment (uriva) [reply gamla/functional.py:172]: thanks

File: gamla/functional.py (hunk 2 above)
Comment (uriva) [reply gamla/functional.py:174]: thanks

**Review by itayzit — APPROVED** [review]

**Pull request merged**